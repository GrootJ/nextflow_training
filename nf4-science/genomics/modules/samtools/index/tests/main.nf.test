nextflow_process {

    name "Test Process SAMTOOLS_INDEX"
    script "../main.nf"
    process "SAMTOOLS_INDEX"

    test("Should index reads_mother.bam correctly") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/bam/reads_mother.bam")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should index reads_father.bam correctly") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/bam/reads_father.bam")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should index reads_son.bam correctly") {

        when {
            params {
                // define parameters here. Example:
                // params.reads_bam = "${projectDir}/data/sample_bams_1.txt"
                outdir = "tests/results"
                // params.outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = Channel.fromPath(params.reads_bam).splitText() // reads_ch = Channel.fromPath(params.reads_bam).splitText()
                input[0] = file("${projectDir}/data/bam/reads_son.bam")
                """
            }
        }

        then {
            assert process.success  // expect the process to run without errors
            assert snapshot(process.out).match()   //expect result of run to be identical to result from a previous run
        }

    }

}
