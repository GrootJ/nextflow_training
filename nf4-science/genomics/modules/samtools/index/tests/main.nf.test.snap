{
    "Should index reads_son.bam correctly": {
        "content": [
            {
                "0": [
                    [
                        "reads_son.bam:md5,af5956d9388ba017944bef276b71d809",
                        "reads_son.bam.bai:md5,a2ca7b84998218ee77eff14af8eb8ca2"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-06T20:02:57.976519"
    },
    "Should index reads_father.bam correctly": {
        "content": [
            {
                "0": [
                    [
                        "reads_father.bam:md5,57f3b977b0eee8278924e38894d02a57",
                        "reads_father.bam.bai:md5,49cb8454781a279a8b33f47209e76fe5"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-06T20:24:39.877633"
    },
    "Should index reads_mother.bam correctly": {
        "content": [
            {
                "0": [
                    [
                        "reads_mother.bam:md5,04b918c3707c24d8f3a7e88bb68e2d32",
                        "reads_mother.bam.bai:md5,1b17cbcc9392e7e787cc83a9081deb31"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-06T20:20:59.044241"
    }
}